#!/usr/bin/env python
# -*- coding: utf-8 -*-
PACKAGE = "offbnode"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Altura_vuelo",     double_t,   0, "En m",      50,  0, 200)
gen.add("Angulo_automatico",bool_t,     0, "Calcular por pendiente de terreno", True)
gen.add("Angulo_entrada",   double_t,   0, "En grados", 45.0,  0, 360)
gen.add("Translape_lateral",double_t,   0, "En %",      70.0,  0, 100)
gen.add("Translape_longitudinal", double_t,   0, "En %",70.0,  0, 100)
gen.add("Turnaround_dist",  double_t,   0, "En m",      20,  0, 100)
gen.add("min_x",            double_t,   0, "En m desde el punto de partida",  -50,  -100, 100)
gen.add("min_y",            double_t,   0, "En m desde el punto de partida",  -50,  -100, 100)
gen.add("max_x",            double_t,   0, "En m desde el punto de partida",  50,  -100, 100)
gen.add("max_y",            double_t,   0, "En m desde el punto de partida",  50,  -100, 100)
gen.add("Ancho_sensor",     double_t,   0, "En mm",     6.17,  0, 10)
gen.add("Altura_sensor",    double_t,   0, "En mm",     4.63,  0, 10)
gen.add("Ancho_imagen_pix", double_t,   0, "En pixeles",4608,  0, 5000)
gen.add("Altura_imagen_pix",double_t,   0, "En pixeles",3456,  0, 5000)
gen.add("Longitud_focal",   double_t,   0, "En mm",     4.9,  0, 10)

exit(gen.generate(PACKAGE, "offbnode", "rute_plan"))